Zotero/Scenes/Detail/ItemDetail/Views/ItemDetailViewController.swift

```diff
@@ -53,36 +69,77 @@ final class ItemDetailViewController: UIViewController {
+                .subscribe(onNext: { [weak viewModel] notification in
+                    guard let viewModel, let notification = notification.object as? AttachmentFileDeletedNotification else { return }
+                    viewModel.process(action: .updateAttachments(notification))
+                })
+                .disposed(by: disposeBag)
+
+            NotificationCenter.default.rx
+                .notification(UIApplication.willEnterForegroundNotification)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] _ in
+                    guard let self else { return }
+                    // Need to reload data to current state before going back to foreground. iOS reloads the collection view when going to foreground with current snapshot. When
+                    // editing text fields we don't update the snapshot (so that the cell is not reloaded while typing), so the edited fields are reset to previous state.
+                    collectionViewHandler.reloadAll(to: viewModel.state, animated: false)
+                })
+                .disposed(by: disposeBag)
+
+            controllers.userControllers?.fileDownloader.observable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] update in
+                    guard let self else { return }
+                    viewModel.process(action: .updateDownload(update))
+
+                    guard viewModel.state.attachmentToOpen == update.key else { return }
+
+                    // 🍎 check logic of processes
+                    switch update.kind {
+                    case .ready:
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachment(key: update.key, parentKey: update.parentKey, libraryId: update.libraryId)
+
+                    case .failed(let error):
+                        viewModel.process(action: .attachmentOpened(update.key))
+                        coordinatorDelegate?.showAttachmentError(error)
+
+                    case .cancelled:
+                        viewModel.process(action: .attachmentOpened(update.key))
+
+                    case .progress:
+                        return
+                    }
+                })
+                .disposed(by: self.disposeBag)
+        }
     }
 
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        self.didAppear = true
+    override func viewIsAppearing(_ animated: Bool) {
+        super.viewIsAppearing(animated)
+        viewModel.process(action: .loadInitialData)
     }
 
     override func viewDidLayoutSubviews() {
         super.viewDidLayoutSubviews()
-
-        if !self.didAppear {
-            // Collapsed abstract is sometimes rendered incorrectly initially. The height of cell has proper height, but only 1 line is shown instead of 2.
-            self.collectionView.reloadData()
```

Comment:
Has this been fixed? By you or iOS update? Can you check whether this still doesn't happen on older supported iOS?

Reply:
Seems ok as far as iOS 15.